### PR TITLE
added type to storage aggregator, with migration

### PR DESCRIPTION
**Pull request by techsmyth**
Type addition to make it easier to track down storage related errors.

Code also uses the type to correctly return the parent entity information, especially for organizations / users

### CODE REVIEW

File: src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts
Comment (valentinyanakiev) [inline src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts]: this is not `NotImplementedException`, more of an `EntityNotFoundException`

File: src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts
Comment (valentinyanakiev) [inline src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts]: this is not NotImplementedException, more of an EntityNotFoundException

File: src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts
Comment (valentinyanakiev) [inline src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts]: this is not NotImplementedException, more of an EntityNotFoundException

File: src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts
Comment (valentinyanakiev) [inline src/services/infrastructure/storage-aggregator-resolver/storage.aggregator.resolver.service.ts]: this is not NotImplementedException, more of an EntityNotFoundException

File: src/migrations/1723036109999-storageAggregatorType.ts
Comment (valentinyanakiev) [inline src/migrations/1723036109999-storageAggregatorType.ts]: I don't see logic about the `Platform` `StorageAggregator`

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Some comments inline.
I also suggest this is covered with API tests before we take it in, then we will make sure it works and it will continue working in the future.

**Comment by techsmyth** [comment]
@valentinyanakiev  comments are I think addressed. Re api test, would not hold this up for that as parentEntity field is not really used outside of spaces for now, so this was more about stopping getting a ton of errors in the logs rather than getting a field we depend on robust.

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
One final change before we merge that - level on `ParentEntity` field resolver returns `SPACE` for `Platform`, `Organization`, `Account`, `User`. That doesn't make any sense - it should be nullable and the field should be null, in case it is not relevant at all for the parent entity.

Raised [a story](https://app.zenhub.com/workspaces/alkemio-development-5ecb98b262ebd9f4aec4194c/issues/gh/alkem-io/test-suites/353) to add tests. Looks to be working functionally with data from acceptance, tested only on API level.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**
